integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/l0promotion/InternalSdxSshAndCmAccessTest.java

```diff
@@ -0,0 +1,144 @@
+import com.sequenceiq.it.util.ResourceUtil;
+import com.sequenceiq.sdx.api.model.SdxClusterStatusResponse;
+
+public class InternalSdxSshAndCmAccessTest extends PreconditionSdxE2ETest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SetWorkloadPasswordAction.class);
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Inject
+    private RecipeTestClient recipeTestClient;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private UmsTestClient umsTestClient;
+
+    @Inject
+    private SshJUtil sshJUtil;
+
+    @Inject
+    private ClouderaManagerUtil clouderaManagerUtil;
+
+    @Test(dataProvider = TEST_CONTEXT)
+    @Description(
+            given = "there is a running Manowar SDX cluster in available state",
+            when = "SSH to the SDX nodes with 'admin' then acting user with new workload password, where the 'post-install' recipe had been installed.",
+            and = "log in to the Cloudera Manager, where 'knox-IDBROKER-BASE' role config mappings then related Cloudera Manager user details have been read",
+            then = "SSH and Cloudera Manager API access should be successful, 'post-install' recipe files and requested Cloudera Manager details should be" +
+                    " present"
+    )
+    public void testSdxSshAndCmAccessWithNewWorkloadPassword(TestContext testContext) throws IOException {
+        String freeIpa = testContext.given(FreeIpaTestDto.class).when(freeIpaTestClient.describe()).getResponse().getName();
+        String sdxInternal = resourcePropertyProvider().getName();
+        String cluster = resourcePropertyProvider().getName();
+        String clouderaManager = resourcePropertyProvider().getName();
+        String recipeName = resourcePropertyProvider().getName();
+        String stack = resourcePropertyProvider().getName();
+        String filePath = "/post-install";
+        String fileName = "post-install";
+        String masterInstanceGroup = "master";
+        String idbrokerInstanceGroup = "idbroker";
+        String newWorkloadPassword = "Admin123@";
+        String workloadUsername = getWorkloadUsername(testContext);
+
+        testContext
+                .given(clouderaManager, ClouderaManagerTestDto.class)
+                .given(cluster, ClusterTestDto.class).withBlueprintName(getDefaultSDXBlueprintName()).withValidateBlueprint(Boolean.FALSE)
+                .withClouderaManager(clouderaManager)
+                .given(RecipeTestDto.class).withName(recipeName).withContent(generateRecipeContent())
+                .withRecipeType(POST_CLOUDERA_MANAGER_START)
+                .when(recipeTestClient.createV4())
+                .given(masterInstanceGroup, InstanceGroupTestDto.class).withHostGroup(MASTER).withNodeCount(1).withRecipes(recipeName)
+                .given(idbrokerInstanceGroup, InstanceGroupTestDto.class).withHostGroup(IDBROKER).withNodeCount(1).withRecipes(recipeName)
```

Comment:
Is necessary to define master and idbroker group count? I think in light duty the nodecount is 1 by default.

Reply:
I'll give a try. Frankly we have not used this on another way yet.

Reply:
I think we do not even SdxInternalTestDto here, SdxTestDto is enough, for example SdxRepairTests uses SdxTestDto without any extra configuration. I see that SshJUtil can handle SdxInternalTestDto only, but SdxTestDto should be enough too to get an IP address for SSH.